src/algebra/power_mod.lean

```diff
@@ -0,0 +1,135 @@
+/-
+Copyright (c) 2019 Neil Strickland. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Neil Strickland
+
+Given a monoid `M` and an element `g ∈ M` satisfying `gⁿ = 1`, we
+can interpret `gⁱ` for elements `i ∈ ℤ/n`.  We give several
+different Lean versions of this idea.  Some of them involve
+the type `zmod n`, which is Lean's implementation of `ℤ/n`.
+Others involve the relation `a ≡ b [MOD n]` on `ℕ`, or the
+relation `a ≡ b [ZMOD n]` on `ℤ`.
+-/
+
+import data.fintype algebra.group_power data.zmod.basic
+
+variable n : ℕ+
+section pow_mod
+
+variables {M : Type*} [monoid M] {g : M} (hg : g ^ n.val = 1)
```

Comment:
I would encourage to use coercions. That way it is easier to interact with the rest of the library. (Simp-lemmas expect coercions, etc...)
Also, for the first couple of lemmas you don't seem to need that `n` is a `pnat`. You could just use `nat`, right?

Reply:
I have switched to using coercions.  There is discussion of `nat` vs `pnat` on [Zulip](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.231076.20powers.20mod.20n).